### ISSUE TITLE
Many iOS users receiving generic stripe error using payment sheet api upon payment intent confirmation

### ISSUE THREAD

**Issue by rednalon** [stripe/stripe-react-native#813]
**Describe the bug**
A number of our users are experiencing issues with processing payments using the payment sheet API on iOS. After attempting card payments using 3d-secure they receive a generic error:
{ 
  "stripeErrorCode": null, 
  "declineCode": null, 
  "localizedMessage": "There was an unexpected error -- try again in a few seconds", 
  "message": "There was an unexpected error -- try again in a few seconds", 
  "type": null, 
  "code": "Failed" 
}

This issue have come up the last few days without any recent modification to our stripe integration. The issue seem to persist while using the same payment method without success after multiple attempts. However, swapping payment method has solved it for some users.

**To Reproduce**
1. Successfully initialize payment sheet with customerId, customerEphemeralKeySecret and paymentIntentClientSecret
2. Successfully add payment option using the payment sheet
3. Attempt to confirm payment by calling confirmPaymentSheetPayment()

**Expected behavior**
Payment attempts using valid payment methods to succeed

**Smartphone (please complete the following information):**
 - Device: any iPhone
 - OS: iOS
 - Version: seemingly any version of iOS

**Additional context**
We have a rather large user base but haven't been able to spot any patterns as to what causes the issue, like:
Card provider
Bank
Device
OS-version

Using @stripe/stripe-react-native: 0.2.3

**Comment by charliecruzan-stripe** [comment]
hey @rednalon , looking into this now. Definitely odd that this started happening without any changes from your side, I'm trying to figure out if there's anything that may have changed on our end to cause this

**Comment by rednalon** [comment]
Hi Charlie,
Good to know you're on it.

No, we have not been able to reproduce it yet on our own devices but will do some more testing soon.

**Comment by lbinscheck** [comment]
@rednalon We also had quite a few users reporting issues with the 3ds authentication. In our case many users experience the 3DS2 flow to be difficult to understand. The UI shows a big button that says "complete" once the 3ds2 sheet opens. And they click it right away. What they actually need to do is to authenticate the payment still, with an app or similar.

TLDR: Check if the paymentIntents use 3DS2 to authenticate the user. 

![IMG_0601](https://user-images.githubusercontent.com/21363655/154307797-fcbca487-29e6-4956-bf3f-51f0db0b2472.PNG)

**Comment by rednalon** [comment]
So, have updated the issue above after managing to narrow down the problem.

**Comment by littleski** [comment]
Any news on this. We have the exact same issue.

**Comment by charliecruzan-stripe** [comment]
@littleski are you also seeing the same behavior (it works when using a `StackNavigator` but not `NativeStackNavigator`)?

**Comment by littleski** [comment]
We use createStackNavigator from @react-navigation. It would be quite painful to try to swtich to something else.
I am not sure this is related to the navigator tho. 3d secure works for most banks, the issue is mainly that the "complete" button is often available for the user to click on it, BEFORE any validation has been performed via the bank app. Thus. If the user "click" on the complete button without having switch to his bank app to do the validation, the 3D secure payment failed. If the user first go to his bank app to validate, and then click complete, it works fine.

**Comment by charliecruzan-stripe** [comment]
I agree that the navigator likely has little to do with the issue, but I believe your issue is separate from the original.

It looks like this could be specifically for out-of-band challenge UIs. If you're experiencing this issue and would like to opt into our testing program, please email [3ds2-mobile-fallback@stripe.com](mailto:3ds2-mobile-fallback@stripe.com) with your [account ID](https://stripe.com/docs/dashboard#find-account-id). Thanks!

**Comment by Djauron** [comment]
> We use createStackNavigator from @react-navigation. It would be quite painful to try to swtich to something else. I am not sure this is related to the navigator tho. 3d secure works for most banks, the issue is mainly that the "complete" button is often available for the user to click on it, BEFORE any validation has been performed via the bank app. Thus. If the user "click" on the complete button without having switch to his bank app to do the validation, the 3D secure payment failed. If the user first go to his bank app to validate, and then click complete, it works fine.

any update ?
